src/adapter/src/coord/sequencer/cluster.rs

```diff
@@ -1010,6 +1010,36 @@ impl Coordinator {
             )
         })
     }
+
+    /// Determine whether we can create a compute item in the specified cluster.
+    ///
+    /// Returns `Ok` if the item can be created, and an error otherwise.
+    pub(crate) fn check_compute_item_in_cluster(
+        &self,
+        name: &QualifiedItemName,
+        cluster_id: ClusterId,
+    ) -> Result<(), AdapterError> {
+        let is_system_schema_specifier = self
+            .catalog()
+            .state()
+            .is_system_schema_specifier(&name.qualifiers.schema_spec);
+
+        let enable_unified_clusters = self
+            .catalog()
+            .for_system_session()
+            .system_vars()
+            .enable_unified_clusters();
+
+        if !is_system_schema_specifier
```

Comment:
Is this line right? Doesn't this imply that we can always create compute items in system schemas?

Reply:
It's a good question! I'm not convinced it's correct now, or before. It was:
```
        if !self
            .catalog()
            .state()
            .is_system_schema_specifier(&name.qualifiers.schema_spec)
            && !self.is_compute_cluster(cluster_id)
        {
```

Spelling this out, I think it comes down to: If the name does not have a system schema specified and is not a compute cluster, then error. The inverse is .. it's ok if it's either a system schema specifier or a compute cluster. I don't know what the system schema specifier is, so I can't say what's correct and what isn't.

Do you know what it _should_ be?

Reply:
Without reading any of the code I think it should be 3 scenarios:

```
if system_schema() {
    Err("Users cannot create objects in system schemas");
} else if !is_compute_cluster() && !enable_unified_clusters {
    Err("BadItemInStorageCluster");
} else {
    Ok(())
}
```

However, the more I think about it, I think 
```
          !self
            .catalog()
            .state()
            .is_system_schema_specifier(&name.qualifiers.schema_spec)
```
is probably there to handle some special edge case. Like introspection indexes are compute objects that are always allowed in storage clusters. Or something like that.

Reply:
If I'm not mistaken, @umanwizard might have added the system schema constraint. Do you remember any details?

Reply:
My memories are very vague now, but I think we wanted to make sure that we could create indexes for introspection objects (e.g. `mz_internal.mz_cluster_replica_metrics`) in storage clusters